src/frontend/packages/lib_classroom/src/components/ClassroomWidgetProvider/widgets/Scheduling/index.tsx

```diff
@@ -0,0 +1,125 @@
+import { Form } from 'grommet';
+import { Maybe } from 'lib-common';
+import {
+  Classroom,
+  debounce,
+  FoldableItem,
+  SchedulingFields,
+} from 'lib-components';
+import React, { useEffect, useRef, useState } from 'react';
+import toast from 'react-hot-toast';
+import { defineMessages, useIntl } from 'react-intl';
+
+import { useUpdateClassroom } from 'data/queries';
+import { useCurrentClassroom } from 'hooks/useCurrentClassroom';
+
+const messages = defineMessages({
+  title: {
+    defaultMessage: 'Scheduling',
+    description: 'A title for the widget.',
+    id: 'component.ClassroomScheduling.title',
+  },
+  info: {
+    defaultMessage: 'This widget can be used to schedule a classroom.',
+    description: 'Helptext for the widget.',
+    id: 'component.ClassroomScheduling.info',
+  },
+  scheduleClassroomLabel: {
+    defaultMessage: 'Update classroom scheduling',
+    description: 'Label for confirm button in classroom scheduling form.',
+    id: 'component.ClassroomScheduling.scheduleClassroomLabel',
+  },
+  updateClassroomSuccess: {
+    defaultMessage: 'Classroom updated.',
+    description: 'Message when classroom update is successful.',
+    id: 'component.ClassroomScheduling.updateClassroomSuccess',
+  },
+  updateClassroomFail: {
+    defaultMessage: 'Classroom not updated!',
+    description: 'Message when classroom update failed.',
+    id: 'component.ClassroomScheduling.updateClassroomFail',
+  },
+});
+
+export const Scheduling = () => {
+  const classroom = useCurrentClassroom();
+  const intl = useIntl();
+  const updateClassroomMutation = useUpdateClassroom(classroom.id, {
+    onSuccess: () => {
+      toast.success(intl.formatMessage(messages.updateClassroomSuccess));
+    },
+    onError: () => {
+      toast.error(intl.formatMessage(messages.updateClassroomFail));
+      setUpdatedClassroomState(classroom);
+    },
+  });
+  const [updatedClassroomState, setUpdatedClassroomState] = useState<
+    Partial<Classroom>
+  >({});
```

Comment:
I don't know if it's important but you are declaring this hook here and you use it in the callback few line before. Moving it before can be a good idea I think